src/java/org/apache/cassandra/service/accord/AccordSerializers.java

```diff
@@ -232,41 +201,45 @@ public Clustering<?> deserialize(DataInputPlus in, int version) throws IOExcepti
         }
 
         @Override
-        public long serializedSize(Clustering<?> clustering, int version)
+        public long serializedSize(Clustering<?> clustering)
         {
             return computeSerializedSize(clustering);
         }
 
         private <V> long computeSerializedSize(Clustering<V> clustering)
         {
-            int size = sizeof(true) + sizeofUnsignedVInt(clustering.size());
```

Comment:
this was a bug found by this refactor.  When we "embed" the bytes we were not validating size == written like we do in networking; when you add that validation you found this was failing